internal/hns/hnsnetwork.go

```diff
@@ -116,20 +258,81 @@ func (network *HNSNetwork) Delete() (*HNSNetwork, error) {
 
 // Creates an endpoint on the Network.
 func (network *HNSNetwork) NewEndpoint(ipAddress net.IP, macAddress net.HardwareAddr) *HNSEndpoint {
-	return &HNSEndpoint{
+	// V1 Api
+	if osversion.Get().Build < osversion.RS5 {
+			return &HNSEndpoint{
+			VirtualNetwork: network.Id,
+			IPAddress:      ipAddress,
+			MacAddress:     string(macAddress),
+		}
+	}
+
+	// V2 Api
+	var endpoint HNSEndpoint{
 		VirtualNetwork: network.Id,
 		IPAddress:      ipAddress,
 		MacAddress:     string(macAddress),
 	}
+	return network.CreateEndpoint(&endpoint)
 }
 
 func (network *HNSNetwork) CreateEndpoint(endpoint *HNSEndpoint) (*HNSEndpoint, error) {
-	operation := "CreateEndpoint"
-	title := "hcsshim::HNSNetwork::" + operation
-	logrus.Debugf(title+" id=%s, endpointId=%s", network.Id, endpoint.Id)
+	// V1 Api
+	if osversion.Get().Build < osversion.RS5 {
+		operation := "CreateEndpoint"
+		title := "hcsshim::HNSNetwork::" + operation
+		logrus.Debugf(title+" id=%s, endpointId=%s", network.Id, endpoint.Id)
```

Comment:
You dont use operation or title. Can you just make this
```go
logrus.Debugf("hcsshim::HNSNetwork::CreateEndpoint, id=%s, endpointid=%s", network.Id, endpoint.Id)
```

Reply:
Done